nservicebus/hosting/selecting.md

```diff
@@ -0,0 +1,78 @@
+This document provides guidance for deciding how to [host](/nservicebus/hosting) your [NServiceBus endpoints](/nservicebus/endpoints/).
+
+This guide does not provide definitive answers for all scenarios, and every decision involves trade-offs. Especially in cloud scenarios, there are currently few definitive answers to what the best solutions are for many scenarios. If it is unclear what the best choice is, or there are very specific constraints, contact [Particular Software](https://particular.net/contactus).
+
+NServiceBus endpoints can be hosted within any .NET process. This guidance groups the hosting options into:
+
+ - [On-premises](#on-premises)
+ - [Containers](#containers)
+ - [Microsoft Azure](#microsoft-azure)
+ - [Amazon Web Services](#amazon-web-services)
+
+## On-premises
+
+For on-premises hosting, endpoints are typically hosted in background process running on servers, which are usually virtual machines. An endpoint can also be hosted in an interactive application, with a user interface, but this guidance focuses on server scenarios.
+
+### Windows Services
+
+In Windows, a [Windows Service](https://docs.microsoft.com/en-us/dotnet/framework/windows-services/introduction-to-windows-service-applications) is the most common way to host NServiceBus endpoints.
+
+The benefits are that Windows Services run in the background, can immediately start when Windows is started, can be paused and restarted, and support [recoverability options](/nservicebus/hosting/windows-service#installation-setting-the-restart-recovery-options-configuring-service-recovery-via-windows-service-properties).
+
+### Internet Information Services (IIS)
+
+In Windows, IIS is a reliable host for [web-based applications](/nservicebus/hosting/web-application). An NServiceBus endpoint can be hosted in any .NET web application, including one running in IIS. However, the purpose of IIS is HTTP request-based hosting. That means IIS will automatically shut down any web application that has not received a request for some time.
+
+This restricts IIS as a choice for hosting NServiceBus endpoints to two specific scenarios:
+
+#### Send-only endpoints
+
+A [send-only](nservicebus/hosting/#self-hosting-send-only-hosting) endpoint is one which sends messages but does not receive any messages. Messages are sent during the handling of incoming HTTP requests or after user input in an interactive application.
+
+The benefit of send-only endpoints is that NServiceBus does not need to perform a lot of checks and initializes much faster.
```

Comment:
@dvdstelt this doesn't feel like the right place to mention the benefits of send-only endpoints. I'd sooner add this sentence to https://docs.particular.net/nservicebus/hosting/#self-hosting-send-only-hosting

Reply:
I'm conflicted, because it would require readers to go through 2 documents to figure out why they'd need SendOnly endpoints when they're hosting inside a webapp. What if we don't call it a benefit, but rather mention something like

> A send-only endpoint is one which sends messages but does not receive any messages. As a result, the endpoint does not need to perform a lot of checks and initializes much faster."

I don't like the "does not need to perform a lot of checks", but couldn't think of anything else, besides "does not need to verify infrastructure" or something.